packages/core/core/test/RequestTracker.test.js

```diff
@@ -307,4 +307,89 @@ describe('RequestTracker', () => {
     assert.strictEqual(cachedResult, 'b');
     assert.strictEqual(called, false);
   });
+
+  it('should write new nodes to cache', async () => {
```

Comment:
Hey we discussed this was the failing test prior to this change.

Could you explain at some point why this line fixes it?

What do you mean re. "nodes getting appended to"?